src/sources/util/tcp.rs

```diff
@@ -162,13 +162,12 @@ fn handle_stream(
     out: impl Sink<SinkItem = Event, SinkError = ()> + Send + 'static,
 ) {
     let mut shutdown = Some(shutdown);
-    let mut token = None;
     let mut reader = FramedRead::new(socket, source.decoder());
     let handler = stream::poll_fn(move || {
         // Gracefull shutdown procedure
         if let Some(future) = shutdown.as_mut() {
             match future.poll() {
-                Ok(Async::Ready(tk)) => {
+                Ok(Async::Ready(_)) => {
```

Comment:
We need to hold onto `ShutdownSignalToken` to have proper shutdown procedure. If we drop it here, we are at the mercy of the other side. Until they close the channel, the whole process will hang on it.